crates/core/tedge_agent/src/agent.rs

```diff
@@ -38,22 +40,35 @@ pub struct AgentConfig {
     pub use_lock: bool,
     pub log_dir: Utf8PathBuf,
     pub data_dir: Utf8PathBuf,
+    pub mqtt_device_topic_id: Arc<str>,
```

Comment:
Using `Arc<str>` here because it's better for immutable data, as it doesn't allocate more memory when cloning, and also tells the reader that it's immutable. More information in [this video](https://www.youtube.com/watch?v=A4cKi7PTJSs).
I think most of the data we're working with in thin-edge is immutable, so I'm experimenting with using these immutable types for new code.

Reply:
Thank you. Indeed a nice hint, that can be used for all the `String`s we use as names or identifiers.

Reply:
This is really nice! It's a great help to distinguish if a member is immutable or mutable by just looking at the struct data type.